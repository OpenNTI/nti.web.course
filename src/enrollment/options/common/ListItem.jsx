import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import {scoped} from '@nti/lib-locale';

const t = scoped('course.enrollment.options.common.ListItem', {
	price: {
		free: 'Free',
		cost: '$%(price)s'
	}
});

export default class CourseEnrollmentOptionsListItem extends React.Component {
	static propTypes = {
		className: PropTypes.string,
		title: PropTypes.string.isRequired,
		price: PropTypes.number,
		selected: PropTypes.bool,
		onSelect: PropTypes.func
	}

	onClick = () => {
		const {onSelect} = this.props;

		if (onSelect) {
			onSelect();
		}
	}

	render () {
		const {title, selected} = this.props;

		return (
			<div className={cx('nti-course-enrollment-option-list-item', {selected})} onClick={this.onClick}>
				<div className="title">{title}</div>
				<div className="meta">
					{this.renderPrice()}
				</div>
			</div>
		);
	}


	renderPrice () {
		const {price} = this.props;

		return (
			<div className="price">
				{price == null ? t('price.free') : t('price.cost', {price})}
			</div>
		);
	}
}
