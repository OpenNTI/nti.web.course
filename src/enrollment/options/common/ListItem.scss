@import "~@nti/style-common";

.nti-course-enrollment-option-list-item {
	cursor: pointer;

	.title {
		font: normal 400 0.875rem/1.3 var(--body-font-family);
		color: var(--primary-grey);
	}

	.meta {
		color: var(--tertiary-grey);
		font: normal 400 0.875rem/1.3 var(--body-font-family);

		.price, .separator, .enrolled {
			display: inline-block;
			vertical-align: middle;
		}

		.separator {
			margin: 0 0.3rem;
		}
	}

	&.selected {
		.meta {
			color: var(--primary-green);
		}
	}
}
