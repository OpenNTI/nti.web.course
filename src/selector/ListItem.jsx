import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import CourseCard from '../card';

export default class CourseSelectorListItem extends React.Component {
	static propTypes = {
		course: PropTypes.object.isRequired,
		onSelect: PropTypes.func,
		selected: PropTypes.bool
	}


	onClick = (e) => {
		e.preventDefault();

		const {course, onSelect} = this.props;

		if (onSelect) {
			onSelect(course);
		}
	}


	render () {
		const {course, selected} = this.props;

		return (
			<CourseCard
				className={cx('nti-course-selector-list-item', {selected})}
				course={course}
				onClick={this.onClick}
				list
			/>
		);
	}
}
