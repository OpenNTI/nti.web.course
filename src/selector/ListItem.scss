@import '~@nti/style-common';

.nti-course-selector-list-item {
	padding: 1.25rem;
	cursor: pointer;
}
