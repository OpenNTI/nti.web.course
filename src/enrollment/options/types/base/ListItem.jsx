import React from 'react';
import PropTypes from 'prop-types';

import ListItem from '../../common/ListItem';

export default class BaseEnrollmentListItem extends React.Component {
	static propTypes = {
		option: PropTypes.shape({
			getPriceDisplay: PropTypes.func,
			getTitle: PropTypes.func.isRequired,
			getPrice: PropTypes.func.isRequired,
			isEnrolled: PropTypes.func.isRequired,
		}),
		selected: PropTypes.bool,
		onSelect: PropTypes.func,
	};

	onSelect = () => {
		const { option, onSelect } = this.props;

		if (onSelect) {
			onSelect(option);
		}
	};

	render() {
		const { option, selected } = this.props;

		return (
			<ListItem
				title={option.getTitle()}
				price={option.getPriceDisplay && option.getPriceDisplay()}
				enrolled={option.isEnrolled()}
				selected={selected}
				onSelect={this.onSelect}
			/>
		);
	}
}
