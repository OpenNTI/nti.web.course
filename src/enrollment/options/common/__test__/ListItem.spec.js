import React from 'react';
import renderer from 'react-test-renderer';

import ListItem from '../ListItem';

/* eslint-env jest */
describe('Course enrollment options common list item ', () => {
	test('Selected with price', async () => {
		const cmp = renderer.create(<ListItem title="my item" price={50} selected/>);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Selected with no price, enrolled', async () => {
		const cmp = renderer.create(<ListItem title="my item" enrolled selected/>);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Title only', async () => {
		const cmp = renderer.create(<ListItem title="my item"/>);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
