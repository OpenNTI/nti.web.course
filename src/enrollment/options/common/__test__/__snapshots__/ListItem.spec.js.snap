// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course enrollment options common list item  Selected with no price, enrolled 1`] = `
<div
  className="nti-course-enrollment-option-list-item selected"
  onClick={[Function]}
>
  <div
    className="title"
  >
    my item
  </div>
  <div
    className="meta"
  >
    <div
      className="price"
    >
      Free
    </div>
    <div
      className="separator"
    >
      -
    </div>
    <div
      className="enrolled"
    >
      Currently Enrolled In
    </div>
  </div>
</div>
`;

exports[`Course enrollment options common list item  Selected with price 1`] = `
<div
  className="nti-course-enrollment-option-list-item selected"
  onClick={[Function]}
>
  <div
    className="title"
  >
    my item
  </div>
  <div
    className="meta"
  >
    <div
      className="price"
    >
      50
    </div>
  </div>
</div>
`;

exports[`Course enrollment options common list item  Title only 1`] = `
<div
  className="nti-course-enrollment-option-list-item"
  onClick={[Function]}
>
  <div
    className="title"
  >
    my item
  </div>
  <div
    className="meta"
  >
    <div
      className="price"
    >
      Free
    </div>
  </div>
</div>
`;
